dhctl/pkg/operations/destroy/destroy.go

```diff
@@ -380,3 +435,101 @@ func (d *ClusterDestroyer) GetMasterNodesIPs(ctx context.Context) ([]NodeIP, err
 
 	return nodeIPs, nil
 }
+
+func (d *StaticMastersDestroyer) switchToNodeuser(settings *session.Session) error {
+	log.DebugLn("Starting replacing SSH client")
+
+	tmpDir := filepath.Join(app.CacheDir, "destroy")
+
+	err := os.MkdirAll(tmpDir, 0o755)
+	if err != nil {
+		return fmt.Errorf("failed to create cache directory for NodeUser: %w", err)
+	}
+
+	log.DebugLn("Tempdir created for SSH client")
+
+	privateKeyPath := filepath.Join(tmpDir, "id_rsa_converger")
+
+	privateKey := session.AgentPrivateKey{
+		Key:        privateKeyPath,
+		Passphrase: d.userCredentials.Password,
+	}
+
+	err = os.WriteFile(privateKeyPath, []byte(d.userCredentials.PrivateKey), 0o600)
+	if err != nil {
+		return fmt.Errorf("failed to write private key for NodeUser: %w", err)
+	}
+
+	log.DebugLn("Private key written")
+
+	if !app.LegacyMode {
+		log.DebugF("Old SSH Client: %-v\n", d.SSHClient)
+		log.DebugLn("Stopping old SSH client")
+		d.SSHClient.Stop()
+
+		// wait for keep-alive goroutine will exit
+		time.Sleep(15 * time.Second)
+	}
+
+	sess := session.NewSession(session.Input{
+		User:           d.userCredentials.Name,
+		Port:           settings.Port,
+		BastionHost:    settings.BastionHost,
+		BastionPort:    settings.BastionPort,
+		BastionUser:    d.userCredentials.Name,
+		ExtraArgs:      settings.ExtraArgs,
+		AvailableHosts: settings.AvailableHosts(),
+		BecomePass:     d.userCredentials.Password,
+	})
+
+	var newSSHClient node.SSHClient
+	if app.LegacyMode {
+		newSSHClient = clissh.NewClient(sess, []session.AgentPrivateKey{privateKey})
+		// Avoid starting a new ssh agent
+		newSSHClient.(*clissh.Client).InitializeNewAgent = false
+	} else {
+		newSSHClient = gossh.NewClient(sess, []session.AgentPrivateKey{privateKey})
+	}
+
+	log.DebugF("New SSH Client: %-v\n", newSSHClient)
+	err = newSSHClient.Start()
+	if err != nil {
+		return fmt.Errorf("failed to start SSH client: %w", err)
+	}
+
+	// adding keys to agent is actual only in legacy mode
+	if app.LegacyMode {
+		err = newSSHClient.(*clissh.Client).Agent.AddKeys(newSSHClient.PrivateKeys())
+		if err != nil {
+			return fmt.Errorf("failed to add keys to ssh agent: %w", err)
+		}
+
+		log.DebugLn("private keys added for replacing kube client")
+	}
+
+	d.SSHClient = newSSHClient
+
+	return nil
+}
+
+func createNodeUser(ctx context.Context, kubeCl client.KubeClient, nodeUser *dhv1.NodeUser) error {
```

Comment:
maybe move this function (or use exists if it is) to pkg/kubernetes/actions/entity package